### DISCUSSION TITLE
GitRepository: change in included repository doesn't seem to trigger an update of the including repository

### DISCUSSION THREAD

**Discussion by exic** [fluxcd/flux2#4662] [Q&A]
Hi,

so I have two GitRepository objects with the same `ref` (one including the other) and a HelmRelease which is using the including repository.

Original repository contains `charts`, `src` and other dirs.
Including GitRepository is configured to ignore all paths except for `charts` dir.
Included GitRepository is configured to ignore all paths except for `src` dir. It is included in a path below charts folder in the including repo.

Say I commit a change in my actual code repo in the `src` dir, flux triggers reconciles for both GitRepository objects as expected. 

However, if flux reconciles the including repository before the included repository, the change is not present in the resulting (including) GitRepository and thus not in the HelmRelease. It will later reconcile the included repo as well, but that is **not triggering an update to the including repository**, though I would expect that. Is it me, my flux version or a bug?

Unfortunately `flux version` gives me `customresourcedefinitions.apiextensions.k8s.io "gitrepositories.source.toolkit.fluxcd.io" is forbidden`, but I assume its the same versions as on another context where I can:
```
flux: v2.2.2
distribution: flux-v2.0.0
helm-controller: v0.35.0
kustomize-controller: v1.0.0
notification-controller: v1.0.0
source-controller: v1.0.0
```

Thanks!

**Comment by souleb** [comment]
I would suggest that you ugrade your cluster and flux cli to the latest version (v2.2.3). Then if the issue persist, run `flux check` and post the result.
